tests/test_logging.py

```diff
@@ -363,3 +363,33 @@ def version(self):
     assert logger.hparams_logged == hparams
     assert logger.metrics_logged != {}
     assert logger.finalized_status == "success"
+
+
+def test_adding_step_key(tmpdir):
+    logged_step = 0
+
+    def validation_end(outputs):
```

Comment:
It would be nice to use Callback instead...

Reply:
AFAIK, there's no easy way to use custom callbacks in `Trainer`, could you provide more details on how to use a callback here?

Reply:
it is true we have in refactoring process right now, #776

Reply:
I don't think this request will bring callback support, as author stated that he does not have enough time to implement it. Hope it'll happen in some other time.
 Still if you have an idea of how to make this test less hacky, I'll be glad to know.

Reply:
that is fine, let's get this done and think about using callback in another PR :]

Reply:
just formatting as it is local func, pls call it `_validation_end`

Reply:
done